util/misc.go

```diff
@@ -72,3 +72,28 @@ func WithRecovery(exec func(), recoverFn func(r interface{})) {
 	}()
 	exec()
 }
+
+// ParseTimeFromPrefix tries to parse time from a prefix of `value`. The formatted time should be separated from
+// other texts. For example, you can parse "20060102-15:04:05 -0700" from a string like "20060102-15:04:05 -0700 FOO"
+// but you can't parse from a string like "20060102-15:04:05 -0700FOO".
+func ParseTimeFromPrefix(format string, value string) (time.Time, error) {
```

Comment:
I think it will be more clear to
1. split `value` by space
2. join the first 2 parts
3. try parse

Reply:
It tries to match the whole string first, because in most cases we can. It's a waste to split it and concat them part by part again, because in most cases we need to concat all parts together to finally get a parseable string.

Reply:
> It tries to match the whole string first
We can keep this part. I was thinking that the two for loops are unnecessary and bring difficulties in code reading.

Reply:
```go
func ParseTimeFromPrefix(format string, value string) (time.Time, error) {
	t, err := time.Parse(format, value)

	if err != nil {
		parts := strings.Split(value, " ")

		for i := len(parts) - 1; i > 0; i-- {
			prefix := strings.Join(parts[:i], " ")
			t, err = time.Parse(format, prefix)
			if err == nil {
				break
			}
		}
	}

	if err != nil {
		err = errors.Errorf("string \"%v\" doesn't has a prefix that matches format \"%v\"", value, format)
	}
	return t, err
}
```
Do  you think this is better?
Actually, IMO, although the original code has two layer for loop, it's simple enough.

Reply:
```go
func ParseTimeFromPrefix(format string, value string) (time.Time, error) {
	t, err := time.Parse(format, value)

	if err != nil {
		parts := strings.Split(value, " ")
		prefix := ""

		for i, part := range parts {
			if i > 0 {
				prefix += " "
			}
			prefix += part
			
			t, err = time.Parse(format, prefix)
			if err == nil {
				break
			}
		}
	}

	if err != nil {
		err = errors.Errorf("string \"%v\" doesn't has a prefix that matches format \"%v\"", value, format)
	}
	return t, err
}
```
Or like this?

Reply:
em, I meant like this:
```
t, err := time.Parse(format, value)
if err != nil {
   if parts := strings.Split(value, " "); len(parts) >= 2 {
      t, err = time.Parse(format, strings.Join(parts[:2], " "))
   }
}
return t, err
```

Reply:
I want to accept a string with arbitrary number of parts. 
Now if we only try to parse the whole string and the first two parts, maybe we'd better to name it something like `CompatibleParseTime`.  
Then the constant `gcTimeFormat` is coupled with this function and should also be moved to this function.
However we can't put this function to gc_worker, because this function is also required in `executor/set.go`, which will cause a cyclic dependency.
Is it ok?